src/main/kotlin/org/jqassistant/tooling/intellij/plugin/data/ReportProviderService.kt

```diff
@@ -0,0 +1,50 @@
+package org.jqassistant.tooling.intellij.plugin.data
+
+import com.buschmais.jqassistant.core.report.api.ReportReader
+import com.buschmais.jqassistant.core.report.impl.XmlReportPlugin
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.project.BaseProjectDirectories.Companion.getBaseDirectories
+import com.intellij.openapi.project.Project
+import com.intellij.openapi.vfs.VirtualFile
+import com.intellij.openapi.vfs.findFile
+import com.intellij.openapi.vfs.isFile
+import org.jqassistant.schema.report.v2.JqassistantReport
+import java.io.File
+
+// Represents an individual report found in the baseDirectory
+data class FoundReport(val baseDirectory: VirtualFile, val report: JqassistantReport)
+
+@Service(Service.Level.PROJECT)
+class ReportProviderService(private val project: Project) {
+    // Returns all found jqassistant reports in the current project
+    // A single project can contain multiple baseDirectories and therefore also multiple report xml files
+    fun readReports(): List<FoundReport> {
+        return project.getBaseDirectories().mapNotNull { baseDir ->
+            // https://plugins.jetbrains.com/docs/intellij/plugin-class-loaders.html
+            val pluginClassLoader = javaClass.classLoader
+            val reader = ReportReader(pluginClassLoader)
+
+            val file = getXmlReportPath(baseDir) ?: return@mapNotNull null
+
+            val report = reader.read(file)
```

Comment:
This might throw an exception, we should handle it I think. Also there is no `@NotNull` annotation on the return value, so we should probably treat it as nullable.